File: src/main/java/org/radarcns/management/security/jwt/algorithm/EcdsaJwtAlgorithm.java
package org.radarcns.management.security.jwt.algorithm;

import com.auth0.jwt.algorithms.Algorithm;

import java.security.KeyPair;
import java.security.interfaces.ECPrivateKey;
import java.security.interfaces.ECPublicKey;

public class EcdsaJwtAlgorithm extends AsymmetricalJwtAlgorithm {
    /** ECDSA JWT algorithm. */
    public EcdsaJwtAlgorithm(KeyPair keyPair) {
        super(keyPair);
        if (!(keyPair.getPrivate() instanceof ECPrivateKey)) {
            throw new IllegalArgumentException(
                    "Cannot make EcdsaJwtAlgorithm with " + keyPair.getPrivate().getClass());
        }
    }

    @Override
    public Algorithm getAlgorithm() {
        return Algorithm.ECDSA256(
                (ECPublicKey)keyPair.getPublic(),
                (ECPrivateKey)keyPair.getPrivate());
    }

    @Override
    public String getEncodedStringHeader() {
        return "-----BEGIN EC PUBLIC KEY-----";
    }

    @Override
    public String getEncodedStringFooter() {
        return "-----END EC PUBLIC KEY-----";
    }
}

File: src/main/java/org/radarbase/management/security/jwt/algorithm/EcdsaJwtAlgorithm.kt
package org.radarbase.management.security.jwt.algorithm

import com.auth0.jwt.algorithms.Algorithm
import java.security.KeyPair
import java.security.interfaces.ECPrivateKey
import java.security.interfaces.ECPublicKey

class EcdsaJwtAlgorithm(keyPair: KeyPair) : AsymmetricalJwtAlgorithm(keyPair) {
    /** ECDSA JWT algorithm.  */
    init {
        require(keyPair.private is ECPrivateKey) { "Cannot make EcdsaJwtAlgorithm with " + keyPair.private.javaClass }
    }

    override val algorithm: Algorithm
        get() = Algorithm.ECDSA256(
            keyPair.public as ECPublicKey,
            keyPair.private as ECPrivateKey
        )
    override val encodedStringHeader: String
        get() = "-----BEGIN EC PUBLIC KEY-----"
    override val encodedStringFooter: String
        get() = "-----END EC PUBLIC KEY-----"
    override val keyType: String
        get() = "EC"
}
